### PR TITLE
Fix #9543 - parent type saves when not using flex relate field

### PR DESCRIPTION
**Pull request by cameronblaikie**
## Description

When saving a flex relate field without saving the parent id it will save the parent type in DB

## Motivation and Context

This has been Assessed internally and has been push live for a client with no issues.
## How To Test This

Example using Notes Module
1. Got to Notes Module.
2. Create a Note and do not relate it to anything and Save.
3. Got to Notes table in the DB and see that the parent_type has not been saved.
4. Do the same again but relate it to a record this time around.
5. Go to Notes table in the DB and see that parent_type has a module and parent_id has a record ID.

For historic data that is effected you will need to run the following SQL Query to fix the data.
`UPDATE moduleName SET parent_type = '' WHERE parent_id = ''`
## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### Final checklist

- [X] My code follows the code style of this project found [here](https://docs.suitecrm.com/community/contributing-code/coding-standards/).
- [ ] My change requires a change to the documentation.
- [X] I have read the [**How to Contribute**](https://docs.suitecrm.com/community/contributing-code/) guidelines.

### CODE REVIEW

**Review by jack7anderson7 — APPROVED** [review]

File: data/SugarBean.php [hunk 1]
```diff
@@ -2397,6 +2397,10 @@ public function save($check_notify = false)
             $this->custom_fields->save($isUpdate);
         }
 
+        if(empty($this->parent_id)) {
```
Comment (clemente-raposo) [inline data/SugarBean.php:2400]: Hi @cameronblaikie @jack7anderson7, I don't think this is the correct place to do this fix. Not all sugar beans have a parent_type and parent_id. Plus, I think that the flex related fields name can differ

**Comment by jack7anderson7** [comment]
Hi $contact_first_name $contact_last_name,
					    We've received your case $acase_name (# $acase_case_number) on $acase_date_entered
					    
					    Status$acase_status
					    Reference$acase_case_number
					    Description$acase_description
					    
On 07/27/2022, 12:10, ***@***.*** ***@***.*** commented on this pull request.



In data/SugarBean.php:
> @@ -2397,6 +2397,10 @@ public function save($check_notify = false)
             $this->custom_fields->save($isUpdate);
         }
 
+        if(empty($this->parent_id)) {

Hi @cameronblaikie @jack7anderson7, I don't think this is the correct place to do this fix. Not all sugar beans have a parent_type and parent_id. Plus, I think that the flex related fields name can differ

**Comment by serhiisamko091184** [comment]
Hello all!
@cameronblaikie 
this PR also requires feedback, please. 
Thanks a lot! 

Regards,
Serhii

**Comment by gunnicom** [comment]
I do not think that this is the correct fix.
Basically you save the value of the dropdown parent_type. The dropdown has a value when you save it, thus making it correct on save.
Maybe just fiving the dropdown the additional opeion of '' would be a better way.

**Comment by serhiisamko091184** [comment]
Hello, @cameronblaikie 

The PR has been marked as stale because there has been no recent activity. It will be closed if no further activity occurs.
Thanks for your contributions.

Regards,
Serhii

**Comment by serhiisamko091184** [comment]
Hey @cameronblaikie 

Closing this PR due to inactivity. 
Thank you.

Regards,
Serhii

**Pull request closed without merging**